### PR TITLE
lazygit: new, 0.43.1

### PR DESCRIPTION
**Pull request by purofle**
Topic Description
-----------------

- lazygit: new, 0.43.1

Package(s) Affected
-------------------

- lazygit: 0.43.1

Security Update?
----------------

No

Build Order
-----------

```
#buildit lazygit
```

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`
- [ ] AArch64 `arm64`
- [ ] LoongArch 64-bit `loongarch64`

**Secondary Architectures**

Architectural progress for secondary ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`
- [ ] RISC-V 64-bit `riscv64`

### CODE REVIEW

File: app-devel/lazygit/spec [hunk 1]
```diff
@@ -0,0 +1,3 @@
+VER=0.43.1
+SRCS="tbl::https://github.com/jesseduffield/lazygit/archive/refs/tags/v$VER.tar.gz"
```
Comment (ziyao233) [inline app-devel/lazygit/spec]: No auto-generated tarball, use git repository instead
See https://wiki.aosc.io/developer/packaging/package-styling-manual/#source-variables

File: app-devel/lazygit/autobuild/build [hunk 2]
```diff
@@ -0,0 +1,10 @@
+abinfo "Building lazygit..."
```
Comment (KexyBiscuit) [inline app-devel/lazygit/autobuild/build]:
```suggestion
abinfo "Building lazygit ..."
```

File: app-devel/lazygit/autobuild/build [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+
+ab_go_build \
+    -trimpath \
+    -mod=readonly
+
+
+abinfo "Installing the binary..."
```
Comment (KexyBiscuit) [inline app-devel/lazygit/autobuild/build]:
```suggestion
abinfo "Installing the binary ..."
```

File: app-devel/lazygit/autobuild/build [hunk 4]
```diff
@@ -0,0 +1,10 @@
+abinfo "Building lazygit..."
+
```
Comment (KexyBiscuit) [inline app-devel/lazygit/autobuild/build]:
```suggestion
```

File: app-devel/lazygit/autobuild/defines [hunk 5]
```diff
@@ -0,0 +1,10 @@
+PKGNAME=lazygit
+PKGSEC=devel
+PKGDES="simple terminal UI for git commands"
```
Comment (KexyBiscuit) [inline app-devel/lazygit/autobuild/defines]:
```suggestion
PKGDES="Simple terminal UI for git commands"
```

File: app-devel/lazygit/spec [hunk 6]
```diff
@@ -0,0 +1,3 @@
...
+CHKSUMS="sha256::a9dad9e5bc9edb1111b3331d1ccb25f97f2593f51b1557a36c1765df08cb3006"
```
Comment (ziyao233) [inline app-devel/lazygit/spec]: An update check entry should be added (`CHKUPDATE="anitya::id=268930"`) [1]
See also [2]: `CHKUPDATE=`

[1]: https://release-monitoring.org/project/268930/
{2]: https://wiki.aosc.io/developer/packaging/package-styling-manual/#source-variables

File: app-devel/lazygit/spec (hunk 1 above)
Comment (KexyBiscuit) [inline app-devel/lazygit/spec]: Consider `SRCS="git::..."`?

File: app-devel/lazygit/autobuild/defines (hunk 5 above)
Comment (ziyao233) [inline app-devel/lazygit/autobuild/defines]: The first alphabet should be capitalized.

File: app-devel/lazygit/autobuild/defines [hunk 7]
```diff
@@ -0,0 +1,10 @@
...
+PKGDEP="git glibc"
```
Comment (KexyBiscuit) [inline app-devel/lazygit/autobuild/defines]: `glibc` is widely available and we don't normally add it to the Depends section.

File: app-devel/lazygit/autobuild/build [hunk 8]
```diff
@@ -0,0 +1,10 @@
...
+ab_go_build \
+    -trimpath \
+    -mod=readonly
+
```
Comment (ziyao233) [inline app-devel/lazygit/autobuild/build]: consider removing this empty line?

**Review by ziyao233 — CHANGES_REQUESTED** [review]

File: app-devel/lazygit/autobuild/defines [hunk 9]
```diff
@@ -0,0 +1,7 @@
+PKGNAME=lazygit
+PKGSEC=devel
```
Comment (KexyBiscuit) [inline app-devel/lazygit/autobuild/defines]: `git` [is](https://github.com/AOSC-Dev/aosc-os-abbs/blob/597a7ab29c0c74560725e86898cf2903996c59f4/app-vcs/git/autobuild/defines#L2) in the `vcs` section, maybe move this package to the same section?
```suggestion
PKGSEC=vcs
```

**Review by KexyBiscuit — CHANGES_REQUESTED** [review]

File: app-devel/lazygit/autobuild/defines (hunk 9 above)
Comment (purofle) [reply app-devel/lazygit/autobuild/defines]: Yes, I have moved the package.

**Review by KexyBiscuit — APPROVED** [review]

File: app-vcs/lazygit/autobuild/defines [hunk 10]
```diff
@@ -0,0 +1,7 @@
+PKGNAME=lazygit
+PKGSEC=vcs
+PKGDES="Simple terminal UI for git commands"
+PKGDEP="git"
+BUILDDEP="go"
+ABINSTALL=lazygit
```
Comment (KexyBiscuit) [inline app-vcs/lazygit/autobuild/defines]: Why is this required?

```suggestion
```

**Review by KexyBiscuit — CHANGES_REQUESTED** [review]

File: app-vcs/lazygit/autobuild/defines (hunk 10 above)
Comment (purofle) [reply app-vcs/lazygit/autobuild/defines]: This is not necessary and has been deleted

File: app-vcs/lazygit/autobuild/build
Comment (xtexx) [inline app-vcs/lazygit/autobuild/build:1]: Try to use ABTYPE=go?

Comment (purofle) [reply app-vcs/lazygit/autobuild/build:1]: The default is ABTYPE=go, but because [build_gomod_install](https://github.com/AOSC-Dev/autobuild4/blob/7421c8b20cb77b64d1fa8883a6f3bb91901c4d66/templates/15-gomod.sh#L50) will also copy the sh files in the project, I used the autobuild/build file

**Review by KexyBiscuit — DISMISSED** [review]
